### PR TITLE
add error decoder for feign client and error decoder tests

### PR DESCRIPTION
**Pull request by mdayican**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/BPS-765

Change description
add error decoder for feign client and error decoder tests

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoderTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        // then
+        assertThat(exception).isInstanceOf(PayHubClientException.class);
+
+        // and
+        PayHubClientException payHubException = (PayHubClientException) exception;
+
+        assertThat(payHubException.getStatus()).isEqualTo(HttpStatus.BAD_REQUEST);
+        assertThat(payHubException.getCause()).isInstanceOf(HttpClientErrorException.class);
+        assertThat(((HttpClientErrorException) payHubException.getCause()).getResponseBodyAsString())
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoderTest.java]: we are not relying on response body - no need to assert it

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoderTest.java]: we are expecting it empty. if later someone would add body to exception, they will not forget to fix test cases?

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoderTest.java]: fare point - fix if implemented later 👍

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoder.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+
+import java.util.Optional;
+
+public class PayHubClientErrorDecoder implements ErrorDecoder {
+
+    private static final Logger log = LoggerFactory.getLogger(PayHubClientErrorDecoder.class);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoder.java]: not that but let's keep it consistent upper case for static final variables

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientErrorDecoder.java]: removed

**Pull request merged**